### PR TITLE
Add bootstrap support for setting up storage pools

### PR DESCRIPTION
**Pull request by wallyworld**
A previous PR #12096 added support for creating Azure encrypted disks from a storage pool and root-disk-source constraint.
However, there was no way to do this for the controller as the storage pool did not exist yet.
This PR allows the creation of a named storage pool at bootstrap, and wires up the controller instance creation to configure the root disk volume params from that storage pool definition. Allowance is made for multiple storage pools, even though right now the CLI can just support the one.
The same arg handling as is used for `--config` is used here. Either a YAML file or multiple cmd args can be used.

## QA steps

```
juju bootstrap azure/australiasoutheast test \
  --no-default-model \
  --storage-pool name=foo \
  --storage-pool type=azure \
  --storage-pool encrypted=true \
  --storage-pool vault-name-prefix=secret \
  --bootstrap-constraints="root-disk-source=foo"
```

Check that the Azure controller VM uses disk encryption.
Check that the controller model has the storage pool created.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1892485

### CODE REVIEW

**Review by manadart — COMMENTED** [review]
Code is good, pending minor changes. Just trying to QA.

File: cmd/juju/commands/bootstrap.go [hunk 1]
```diff
@@ -1228,6 +1250,7 @@ type bootstrapConfigs struct {
 	bootstrap                bootstrap.Config
 	inheritedControllerAttrs map[string]interface{}
 	userConfigAttrs          map[string]interface{}
+	storagePools             map[string]map[string]interface{}
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply cmd/juju/commands/bootstrap.go]: done

File: state/initialize.go [hunk 2]
```diff
@@ -430,3 +439,28 @@ func (st *State) createDefaultStoragePoolsOps(registry storage.ProviderRegistry)
...
+		}
+		providerType := storage.ProviderType(pType)
+		if _, err := pm.Create(name, providerType, attrs); err != nil {
+			return nil, errors.Annotatef(
+				err, "adding custom storage pool %q", name,
+			)
+		}
+	}
+
+	logger.Criticalf("SP: %#v", m.Settings)
```
[the remark this answers is not in the document]
Comment (wallyworld) [reply state/initialize.go]: doh

**Review by manadart — APPROVED** [review]
QA worked fine.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**